### PR TITLE
Tier-1 test cases for KMS workflow

### PR DESCRIPTION
**Pull request by GowthamShanmugam**

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
@bipuladh updated this Pr

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/helpers/vault.ts [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+      }
+    }
+}
+
+export type vaultConfigData = {
+    VAULT_TOKEN: string,
+    VAULT_HOST_ADDRESS: string,
+    VAULT_PORT: string,
+    VAULT_SECRETS_PATH: string,
+}
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/helpers/vault.ts:313]: Add a new line

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts [hunk 2]
```diff
@@ -71,8 +77,15 @@ Cypress.Commands.add('install', (mode: 'Internal' | 'Attached' = 'Internal', enc
 
       // Step 2
       if (encrypted) {
-        cy.log('Enabling Encryption');
-        cy.byTestID('encryption-checkbox').click();
+          // Enable encryption
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts:80]:
```suggestion
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                    cy.exec(`oc exec  -ti ${podName} -- /bin/sh -c 'export VAULT_TOKEN=${vaultToken} &&  vault secrets enable -path=secret kv'`); 
+                    cy.log(`vault token = ${vaultToken}`);
+                    cy.exec(`kubectl create secret generic vault-token --from-literal=token=${vaultToken} -n hashicorp`);
+                });
+            });
+        } else {
+            cy.log("Vault is already deployed");
+        }
+    });
+};
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts:64]: add new line

**Review by bipuladh — CHANGES_REQUESTED** [review]
Initial review

**Comment by bipuladh** [comment]
/hold
Moved out to 4.8.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/helpers/vault.ts (hunk 1 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/helpers/vault.ts:313]: ack

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts (hunk 2 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts:80]: ack

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts (hunk 3 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts:64]: Ack

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      cy.exec(`echo '${JSON.stringify(serviceAccountJSON)}' | oc apply -f -`);
+      cy.exec(`echo '${JSON.stringify(roleBindingJSON)}' | oc apply -f -`);
+      cy.exec(`echo '${JSON.stringify(getPVCJSON)}' | oc apply -f -`);
+      cy.exec(`echo '${JSON.stringify(configMapJSON)}' | oc apply -f -`);
+
+      cy.log('Deploying vault');
+      cy.exec(`echo '${JSON.stringify(deploymentJson)}' | oc apply -f -`);
+      cy.exec(`echo '${JSON.stringify(serviceJSON)}' | oc apply -f -`);
+      cy.log('Waiting for vault pod to be up and running');
+      cy.wait(50000);
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts:31]: Deploying vault in the background, Need this wait to talk with vault server, Frontend check failure is expected

**Comment by bipuladh** [comment]
We are in 4.8 removing hold.
/hold cancel

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts (hunk 4 above)
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts:31]: perform polling on it.

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts]: ack

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  testDeploymentJSON,
+} from '../helpers/vault';
+import { commandPoll } from '../views/common';
+
+export const configureVault = () => {
+  cy.exec('oc get project hashicorp', {
+    failOnNonZeroExit: false,
+  }).then(({ code }) => {
+    // Deploy vault only if doesn't already exist
+    if (code !== 0) {
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts:19]:
```suggestion
```
Is there a reason why we can't assume vault doesn't exist?
Are multiple tests using this?

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/vault-standalone.ts:19]: Right now only my test cases are using, in the future, we may add KMS cluster-wide encryption also. Also, while running test cases locally it helps a lot. It saves time.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/storage-class.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  // select block pool
+  cy.byTestID('pool-dropdown-toggle').click();
+  cy.byTestID('ocs-storagecluster-cephblockpool').click();
+
+  // encryption
+  encrypted && configureKms();
+
+  // create new sc
+  cy.byTestID('create-action').click();
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/storage-class.ts:38]:
```suggestion
  cy.byTestID('create-action').click();
```
Could we use `type=submit` here?

File: frontend/public/components/storage-class-form.tsx [hunk 7]
```diff
@@ -1006,6 +1006,7 @@ class StorageClassFormWithTranslation extends React.Component<
               id="storage-class-name"
               onChange={(event) => this.setStorageHandler('name', event.target.value.trim())}
               value={_.get(newStorageClass, 'name', '')}
+              data-test="storage-class-name-text"
```
Comment (bipuladh) [inline frontend/public/components/storage-class-form.tsx:1009]: The form above has a data-test-id could we perhaps do something like
`cy.getById('storage-class-form').get('input#storage-class-name')`.

File: frontend/public/components/storage-class-form.tsx [hunk 8]
```diff
@@ -1094,6 +1094,7 @@ class StorageClassFormWithTranslation extends React.Component<
                 onClick={this.createStorageClass}
                 type="submit"
                 variant="primary"
+                data-test="create-action"
```
Comment (bipuladh) [inline frontend/public/components/storage-class-form.tsx:1097]: Did you miss this? We could use instead `button[type="submit"]` where it is required?

**Review by bipuladh — CHANGES_REQUESTED** [review]

**Comment by GowthamShanmugam** [comment]
@bipuladh all changes are done

**Pull request merged**